solidity/contracts/MezoAllocator.sol

```diff
@@ -225,20 +231,32 @@ contract MezoAllocator is IDispatcher, Ownable2StepUpgradeable {
     function withdraw(uint256 amount) external {
         if (msg.sender != address(stbtc)) revert CallerNotStbtc();
 
-        emit DepositWithdrawn(depositId, amount);
+        uint256 unallocatedBalance = tbtc.balanceOf(address(this));
+
+        if (amount > unallocatedBalance) {
+            uint256 amountToWithdraw = amount - unallocatedBalance;
+
+            emit DepositWithdrawn(depositId, amountToWithdraw);
+
+            if (amountToWithdraw < depositBalance) {
+                mezoPortal.withdrawPartially(
+                    address(tbtc),
+                    depositId,
+                    uint96(amountToWithdraw)
+                );
+            } else if (amountToWithdraw > depositBalance) {
```

Comment:
Not a blocker and nothing wrong, just a heads up for future changes. In `Portal` there is a way someone can [donate](https://github.com/thesis/mezo-portal/blob/f563927ee6c5b670599bd7d0973194b5537b3c4f/solidity/contracts/Portal.sol#L669) your deposit. In this case, it may happen `depositBalance` tracked separately will be lower than what's actually in the Portal.